### PR TITLE
Custom Query Endpoint

### PR DESCRIPTION
**Pull request by dhercher**
Adding support for schedule_query via the new v2/consolidation endpoint.  

The endpoint accepts the same options as the old endpoint except that it currently requires the zk deployment name to be submitted

### CODE REVIEW

File: alooma/alooma.py [hunk 1]
```diff
@@ -60,6 +60,7 @@
 MAPPING_TIMEOUT = 300
 
 BASE_URL = 'https://app.alooma.com'
+CUSTOM_CONSOLIDATION_V2 = 'v2/consolidation/custom'
```
[the remark this answers is not in the document]
Comment (dhercher) [reply alooma/alooma.py]: correct

File: alooma/alooma.py [hunk 2]
```diff
@@ -1324,15 +1328,19 @@ def schedule_query(self, event_type, query, frequency=None, run_at=None):
...
-            "custom_query": query,
-            "event_type": event_type,
-            "frequency": frequency,
-            "run_at": run_at
+        "custom_query": query,
+        "event_type": event_type,
+        "deployment_name": self.get_zk_deployment_name(),
+        "avoid_duplicates": False
         }
+        if frequency:
```
[the remark this answers is not in the document]
Comment (dhercher) [reply alooma/alooma.py]: We can do that but idk how you could submit and change this logic + it has to be here for backwards compat.  Do you just mean adding functions for these as well?

[the remark this answers is not in the document]
Comment (oxmane) [reply alooma/alooma.py]: You are correct
Didn't think of backwards compatibility :(

File: alooma/alooma.py [hunk 3]
```diff
@@ -1324,15 +1328,19 @@ def schedule_query(self, event_type, query, frequency=None, run_at=None):
...
 
         # Prep Data for Consolidation Post
         data = {
-            "custom_query": query,
-            "event_type": event_type,
-            "frequency": frequency,
-            "run_at": run_at
+        "custom_query": query,
+        "event_type": event_type,
+        "deployment_name": self.get_zk_deployment_name(),
```
[the remark this answers is not in the document]
Comment (shani-einav) [reply alooma/alooma.py]: @oxmane I did not worked on it

**Pull request merged**